rust/worker/src/cache/log_cache.rs

```diff
@@ -0,0 +1,242 @@
+use crate::cache::config::EvictionConfig;
+use crate::cache::config::LogCacheConfig;
+use crate::types::LogRecord;
+use ahash::RandomState;
+use foyer::CacheContext;
+use foyer::LruConfig;
+use foyer_memory::Cache;
+use foyer_memory::CacheBuilder;
+use foyer_memory::CacheEntry;
+use foyer_memory::CacheEventListener;
+use foyer_memory::DefaultCacheEventListener;
+use once_cell::sync::OnceCell;
+use std::cmp::Ord;
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+
+static LOG_CACHE: OnceCell<LogCache> = OnceCell::new();
+
+pub fn init(config: &LogCacheConfig) {
+    if LOG_CACHE.set(LogCache::new(&config)).is_err() {
+        panic!("log cache already initialized");
+    }
+}
+
+pub fn get() -> &'static LogCache {
+    LOG_CACHE.get().expect("log cache not initialized")
+}
+
+#[derive(Eq, PartialEq, Hash, Debug)]
+pub(crate) struct LogCacheKey {
+    pub(crate) collection_id: String,
+    pub(crate) start_log_offset: i64,
+}
+
+impl Ord for LogCacheKey {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.collection_id
+            .cmp(&other.collection_id)
+            .then(self.start_log_offset.cmp(&other.start_log_offset))
+    }
+}
+
+impl PartialOrd for LogCacheKey {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+pub(crate) type LogCacheEntry = CacheEntry<LogCacheKey, Arc<[LogRecord]>, LogCacheEventListener>;
+
+pub(crate) struct LogCacheEventListener {}
+
+impl CacheEventListener<LogCacheKey, Arc<[LogRecord]>> for LogCacheEventListener {
+    fn on_release(
+        &self,
+        key: Arc<LogCacheKey>,
+        value: Arc<Arc<[LogRecord]>>,
+        context: CacheContext,
+        charges: usize,
+    ) {
+        println!(
+            "LogCacheEventListener::on_release: key: {:?}, value: {:?}, context: {:?}, charges: {}",
+            key, value, context, charges
+        );
+    }
+}
+
+pub(crate) struct LogCache {
+    cache: Cache<LogCacheKey, Arc<[LogRecord]>, LogCacheEventListener>,
```

Comment:
Please use `foyer::Cache` is only in-memory is needed, otherwise, use `foyer::HybridCache` instead.

In short, only `foyer` needs to be imported.